File: python/algorithms/easy/timeConversion.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Time Conversion - HackerRank
Test description: https://www.hackerrank.com/challenges/time-conversion/problem
"""
#---------- functions ----------
def timeConversion(s: str) -> str:
    """
    Converts a time in 12-hour AM/PM format to military (24-hour) time.

    Args:
        s (str): A string representing the time in 12-hour format (e.g., "07:05:45PM").

    Returns:
        str: A string representing the time in 24-hour format (e.g., "19:05:45").
    """
    hour = int(s[0:2])
    minute = s[3:5]
    second = s[6:8]
    period = s[8:]

    if period == "AM":
        if hour == 12:
            hour = 0
    else:
        if hour != 12:
            hour += 12

    return f"{hour:02}:{minute}:{second}"

File: java/algorithms/easy/timeConversion.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-13
 * Last modified: 2026-06-13
 * @description Staircase - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/staircase/problem}
 */
//---------- solution ----------
class Result
{
    /**
     * Converts a time in 12-hour AM/PM format to military (24-hour) time.
     * @param {string} s
     * @return {string}
     */
    public static String timeConversion(String s) 
    {
        int hour = Integer.parseInt(s.substring(0, 2));
        String minute = s.substring(3, 5);
        String second = s.substring(6, 8);
        String period = s.substring(8);

        if (period.equals("AM"))
        {
            if (hour == 12)
            { hour = 0; }
        }
        else
        {
            if (hour != 12) 
            { hour += 12; }
        }//end if

        return (String.format("%02d:%s:%s", hour, minute, second));
    }//end timeConversion()
}//end class